docs/Solana/DataFeeds/using-data-feeds-off-chain.md

```diff
@@ -62,34 +80,150 @@ After you install the required tools, clone the example code from the [solana-st
+
+1. Install [Node.js 14 or higher](https://nodejs.org/en/download/). Run `node --version` to verify which version you have installed:
+
+    ```sh
+    node --version
+    ```
+
+1. Change to your project directory or create a new directory.
+
+    ```sh
+    mkdir off-chain-project && cd off-chain-project
+    ```
+
+1. Optionally [install Yarn](https://classic.yarnpkg.com/lang/en/docs/install/) to use as a package manager and initialize yarn if your project does not already have a `package.json` file:
+
+    ```sh
+    npm install -g yarn && yarn init
+    ```
+
+1. Add the [Anchor library](https://www.npmjs.com/package/@project-serum/anchor) to your project:
+
+    ```sh yarn
+    yarn add @project-serum/anchor
+    ```
+    ```sh npm
+    npm i @project-serum/anchor
+    ```
+
+1. Add the [Chainlink Solana NPM library](https://www.npmjs.com/package/@chainlink/solana-sdk) to your project:
+
+    ```sh yarn
+    yarn add @chainlink/solana-sdk
+    ```
+    ```sh npm
+    npm i -g @chainlink/solana-sdk
+    ```
+
+1. Create a temporary Solana wallet to use for this example. Alternatively, if you have an existing wallet that you want to use, locate the path to your [keypair](https://docs.solana.com/terminology#keypair) file and use it as the keypair for the rest of this guide.
+
+    ```sh
+    solana-keygen new --outfile ./id.json
+    ```
+
+1. Set the [Anchor environment variables](https://www.twilio.com/blog/2017/01/how-to-set-environment-variables.html). Anchor uses these to determine which wallet to use and how to get a connection to a Solana cluster. Because this example does not generate or sign any transactions, no SOL tokens are required. The wallet is required only by the Anchor library. For a list of available networks and endpoints, see the [Solana Cluster RPC Endpoints](https://docs.solana.com/cluster/rpc-endpoints) documentation.
+
+    ```sh Solana Devnet
+    export ANCHOR_PROVIDER_URL=https://api.devnet.solana.com &&
+    export ANCHOR_WALLET=./id.json
+    ```
+    ```sh Solana Mainnet
+    export ANCHOR_PROVIDER_URL=https://api.mainnet-beta.solana.com &&
+    export ANCHOR_WALLET=./id.json
+    ```
+    ```sh Solana Mainnet Project Serum Endpoint
+    export ANCHOR_PROVIDER_URL=https://solana-api.projectserum.com &&
+    export ANCHOR_WALLET=./id.json
     ```
 
+1. Download the sample code. This example queries price data off-chain. By default, the script reads the SOL/USD feed, but you can change the `CHAINLINK_FEED_ADDRESS` variable to point to any of the  value with the [feed account](https://docs.chain.link/docs/solana/data-feeds-solana/) that you wish to query. Alternatively, you can copy the code sample below.
+
+    ```sh JavaScript
+    curl -o off-chain-read.js https://docs.chain.link/samples/Solana/PriceFeeds/off-chain-read.js
+    ```
+    ```sh TypeScript
+    curl -o off-chain-read.ts https://docs.chain.link/samples/Solana/PriceFeeds/off-chain-read.ts
+    ```
```

Comment:
Yes I'm thinking why are we linking to actual files in the 'add to existing project' section? Because devs who will be here have projects they want to add code to, hence why I originally had just the sample code they can copy/paste directly from the docs. Is there a specific reason they were also put in files, and the files were prioritized in the order? I don't think devs will want to run curl commands on scripts, they will want to copy & paste code into their project and then test there

Reply:
I simplified this step and re-prioritized the code samples. As long as we also provide commands to help people run these examples, they can still reach a success state. The goal is for most users to have a fast path to something that runs so that they realize how easy it is and take the next step towards integration.